tests/utils/shippable/shippable.sh

```diff
@@ -0,0 +1,156 @@
+else
+    # on-demand coverage reporting disabled (default behavior, always-on coverage reporting remains enabled)
+    export COVERAGE="--coverage-check"
+fi
+
+if [ -n "${COMPLETE:-}" ]; then
+    # disable change detection triggered by setting the COMPLETE environment variable to a non-empty value
+    export CHANGED=""
+elif [[ "${COMMIT_MESSAGE}" =~ ci_complete ]]; then
+    # disable change detection triggered by having 'ci_complete' in the latest commit message
+    export CHANGED=""
+else
+    # enable change detection (default behavior)
+    export CHANGED="--changed"
+fi
+
+if [ "${IS_PULL_REQUEST:-}" == "true" ]; then
+    # run unstable tests which are targeted by focused changes on PRs
+    export UNSTABLE="--allow-unstable-changed"
+else
+    # do not run unstable tests outside PRs
+    export UNSTABLE=""
+fi
+
+virtualenv --python /usr/bin/python3.7 ~/ansible-venv
+set +ux
+. ~/ansible-venv/bin/activate
+set -ux
+
+#pip install ansible==2.9.0 --disable-pip-version-check
+pip install git+https://github.com/ansible/ansible.git@temp-2.10-devel --disable-pip-version-check
+
+COLLECTION_DIR="${HOME}/.ansible/ansible_collections/"
+TEST_DIR="${COLLECTION_DIR}/community/rabbitmq"
+mkdir -p "${TEST_DIR}"
+cp -aT "${SHIPPABLE_BUILD_DIR}" "${TEST_DIR}"
+cd "${TEST_DIR}"
+
+function cleanup
+{
+    if [ -d tests/output/coverage/ ]; then
+        if find tests/output/coverage/ -mindepth 1 -name '.*' -prune -o -print -quit | grep -q .; then
+            # for complete on-demand coverage generate a report for all files with no coverage on the "other" job so we only have one copy
+            if [ "${COVERAGE}" == "--coverage" ] && [ "${CHANGED}" == "" ] && [ "${test}" == "sanity/1" ]; then
+                stub="--stub"
+            else
+                stub=""
+            fi
+
+            # shellcheck disable=SC2086
+            ansible-test coverage xml --color -v --requirements --group-by command --group-by version ${stub:+"$stub"}
+            cp -a tests/output/reports/coverage=*.xml shippable/codecoverage/
+        fi
+    fi
+
+    if [ -d  tests/output/junit/ ]; then
+      cp -aT tests/output/junit/ shippable/testresults/
+    fi
+
+    if [ -d tests/output/data/ ]; then
+      cp -a tests/output/data/ shippable/testresults/
+    fi
+
+    if [ -d  tests/output/bot/ ]; then
+      cp -aT tests/output/bot/ shippable/testresults/
+    fi
+}
+
+trap cleanup EXIT
+
+if [[ "${COVERAGE:-}" == "--coverage" ]]; then
+    timeout=60
+else
+    timeout=45
+fi
+
+# STAR: HACK install dependencies
+(
+#ansible.windows doesn't install from Galaxy
+mkdir /tmp/collection_deps
+git clone https://github.com/ansible-collections/ansible.windows.git /tmp/collection_deps/ansible.windows
+cd /tmp/collection_deps/ansible.windows
+ansible-galaxy collection build
+ansible-galaxy collection install /tmp/collection_deps/ansible.windows/ansible-windows* -p "${COLLECTION_DIR}"
+)
+
+
+#ansible-galaxy collection install ansible.windows -p "${COLLECTION_DIR}"
```

Comment:
Most of the others can do, though need to check for any Python or integration/unit tests imports

Reply:
Are these installs really required? It doesn't seem to me like these are used anywhere in this collection.